apps/api/v1/pages/api/event-types/_post.ts

```diff
@@ -275,24 +275,57 @@ async function postHandler(req: NextApiRequest) {
     children: _,
     ...parsedBody
   } = schemaEventTypeCreateBodyParams.parse(body || {});
+  if (parsedBody.teamId && parsedBody.parentId) {
+    //parentId and teamId can't be used together as parentId is present only for child event types of a managed event
+    //teamId is used for event types belonging to a team
+    //child events of a team event do not have teamId , they are identified via parentId
+    //throw error to avoid unexpected behaviour
+    throw new HttpError({
+      statusCode: 400,
+      message: "`parentId` and `teamId` both cannot be present in the request",
+    });
+  }
+  if (parsedBody.teamId && parsedBody.userId) {
+    //team event is not associated with an owner or user
+    throw new HttpError({
+      statusCode: 400,
+      message: "`teamId` and `userId` both cannot be present in the request",
+    });
+  }
+  if (!parsedBody.teamId && parsedBody.schedulingType) {
+    //schedulingType is applicable only for team events
+    throw new HttpError({
+      statusCode: 400,
+      message: "schedulingType is applicable only for team events",
+    });
+  }
 
   let data: Prisma.EventTypeCreateArgs["data"] = {
     ...parsedBody,
-    userId,
-    users: { connect: { id: userId } },
     bookingLimits: bookingLimits === null ? Prisma.DbNull : bookingLimits,
     durationLimits: durationLimits === null ? Prisma.DbNull : durationLimits,
   };
 
+  if (!parsedBody.teamId) {
+    //connect user if eventtype doesn't belong to a team
+    data = {
+      ...data,
+      userId,
+      users: { connect: { id: userId } },
+    };
+  }
+
   await checkPermissions(req);
 
-  if (parsedBody.parentId) {
+  //user with admin or owner role on team can create child event types
+  if (parsedBody.parentId && !isSystemWideAdmin) {
     await checkParentEventOwnership(req);
     await checkUserMembership(req);
   }
 
-  if (isSystemWideAdmin && parsedBody.userId) {
-    data = { ...parsedBody, users: { connect: { id: parsedBody.userId } } };
+  //checks for owner/admin role on team have been verified above if parentId is present
+  if ((isSystemWideAdmin || parsedBody.parentId) && parsedBody.userId) {
```

Comment:
I think it should it be  `if (isSystemWideAdmin || (parsedBody.parentId && parsedBody.userId))` instead

Reply:
also to avoid having to check for systemWideAdmin in multiple place I would simple have the check at the top and return as early as possible if user is system wide admin, so we don't need to worry about it in the rest of the function

Reply:
`
if (isSystemWideAdmin || (parsedBody.parentId && parsedBody.userId)) `

--> this would unconditionally attach userId ( which may be undefined ) 

I think it would be necessary to have userId check to prevent attaching undefined value

Reply:
this if case is too unclear in this case, I would refactor to reduce the complexity of this